### ISSUE TITLE
Investigate whether the scraper convention of returning -1 on failure and None on success is consistent with GAE, or arbitrary

### ISSUE THREAD

**Issue by ohinds-vera** [Recidiviz/recidiviz-data#114]
And if it's arbitrary, consider refactoring for a convention of throwing exceptions on errors and removing the explicit `return None` statements.

**Comment by ohinds-vera** [comment]
When resolving this issue, make sure to update all children of Scraper, as well as Scraper itself.

**Comment by jessex** [comment]
I agree in principal. What I need to look at in the code again is whether there are places where we are using -1 to mean non-exceptional in addition to exceptional. I think we should throw exceptions where something is exceptional and let it bubble to the right place in the stack (which may be all the way up to the request handler) and I think we should _not_ throw exceptions if it's purely as a means of control flow. In those cases, having some return value (maybe None) is sensible. Probably this isn't worth prioritizing until we've hit a steadier state in the ingest architecture, e.g. once we've gotten the persistence and database layers into place, and we've gotten monitoring into place.

+CC @arian487
